unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GLTFShape/Handler/GLTFShapeComponentHandler.cs

```diff
@@ -0,0 +1,125 @@
+                ApplyModel(model);
+            
+            this.model = model;
+        }
+                
+        public bool IsVisible() {  return model.Visible; }
+        
+        public bool HasCollisions() {  return model.WithCollisions; }
+
+        private void LoadShape(IParcelScene scene, IDCLEntity entity, PBGLTFShape model)
+        {
+            ContentProvider provider = scene.contentProvider;
+            if (provider.HasContentsUrl(model.Src))
+            {
+                entity.EnsureMeshGameObject("GLTF mesh");
+
+                if (loadWrapper != null)
+                {
+                    loadWrapper.Unload();
+                    DisposeMesh(scene);
+                }
+                // We prepare a load wrapper to load the GLTF
+                loadWrapper = new LoadWrapper_GLTF();
+                loadWrapper.customContentProvider = provider;
+    
+                entity.meshesInfo.currentShape = this;
+
+                loadWrapper.entity = entity;
+                loadWrapper.useVisualFeedback = Configuration.ParcelSettings.VISUAL_LOADING_ENABLED;
+                loadWrapper.initialVisibility = true;
+                loadWrapper.Load(model.Src, (wrapper) =>
+                {
+                    // We remove the transition from the GLTF
+                    if (!model.Visible)
+                    {
+                        MaterialTransitionController[] materialTransitionControllers = entity.meshRootGameObject.GetComponentsInChildren<MaterialTransitionController>();
+
+                        for (var i = 0; i < materialTransitionControllers.Length; i++)
+                            GameObject.Destroy(materialTransitionControllers[i]);
+                    }
+                    entity.meshesInfo.meshRootGameObject = entity.meshRootGameObject;
+                    meshesInfo = entity.meshesInfo;
+                    
+                    // Apply the model for visibility, collision and event pointer
+                    ApplyModel(model);
+                    dataStore.RemovePendingResource(scene.sceneData.id, model);
+                    
+                }, (wrapper, exception) =>
+                {
+                    dataStore.RemovePendingResource(scene.sceneData.id, model);
+                });
+            }
+            else
+            {
+#if UNITY_EDITOR
+                Debug.LogWarning($"LoadableShape '{model.Src}' not found in scene '{scene.sceneData.id}' mappings");
+#endif
+            }
+            dataStore.AddPendingResource(scene.sceneData.id, model);
+        }
+        
+        internal void ApplyModel(PBGLTFShape model)
+        {
+            // Set visibility
+            meshesInfo.meshRootGameObject.SetActive(model.Visible);
+            
+            // Set collisions
+            foreach (var collider in meshesInfo.colliders)
+            {
+                collider.enabled = model.WithCollisions;
+            }
+            
+            //TODO: Implement events here
```

Comment:
What events?

Reply:
I will change the comment for a better explanation. Long story short, the events for when you click on the entities, those events needs to be implemented again in ECS7